src/RemotePullManager.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace Drupal\contentpool_client;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\State\StateInterface;
+use Drupal\multiversion\Workspace\ConflictTrackerInterface;
+use Drupal\relaxed\Entity\Remote;
+use Drupal\replication\Entity\ReplicationLogInterface;
+use Drupal\workspace\ReplicatorInterface;
+
+/**
+ * Helper class to get training references and backreferences.
+ */
+class RemotePullManager implements RemotePullManagerInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The state service.
+   *
+   * @var \Drupal\Core\State\StateInterface
+   */
+  protected $state;
+
+  /**
+   * The replicator manager.
+   *
+   * @var \Drupal\workspace\ReplicatorInterface
+   */
+  protected $replicatorManager;
+
+  /**
+   * The injected service to track conflicts during replication.
+   *
+   * @var \Drupal\multiversion\Workspace\ConflictTrackerInterface
+   */
+  protected $conflictTracker;
+
+  /**
+   * Constructs a RemoteAutopullManager object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, StateInterface $state, ReplicatorInterface $replicator_manager, ConflictTrackerInterface $conflict_tracker) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->state = $state;
+    $this->replicatorManager = $replicator_manager;
+    $this->conflictTracker = $conflict_tracker;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function pullAllRemotes() {
+    $remotes = $this->entityTypeManager->getStorage('remote')->loadMultiple();
+
+    $counter = 0;
+    foreach ($remotes as $remote) {
+      // We try to do a pull from the remote.
+      $this->doPull($remote);
```

Comment:
this does not check the "autopull" flag for the remote, but it should - if the user has non-contentpool remotes they should not be pulled?